contract-examples/contracts/ExampleRewardManager.sol

```diff
@@ -31,3 +34,107 @@ contract ExampleRewardManager is Ownable {
     return rewardManager.areFeeRecipientsAllowed();
   }
 }
+
+contract ExampleRewardManagerTest is DSTest {
+  uint blackholeBalance;
+  ExampleRewardManager exampleReceiveFees;
+  uint exampleBalance;
+
+  function setUp() public {
+    blackholeBalance = BLACKHOLE_ADDRESS.balance;
+  }
+
+  function test_sendFeesToBlackhole() public {
+    blackholeBalance = BLACKHOLE_ADDRESS.balance;
+  }
+
+  function test_checkSendFeesToBlackhole() public {
+    assertGt(BLACKHOLE_ADDRESS.balance, blackholeBalance);
+  }
+
+  function test_doesNotSetRewardAddressBeforeEnabled() public {
+    ExampleRewardManager example = new ExampleRewardManager();
+    address exampleAddress = address(example);
+    IRewardManager rewardManager = IRewardManager(REWARD_MANAGER_ADDRESS);
+
+    assertEq(rewardManager.readAllowList(exampleAddress), 0);
+
+    try example.setRewardAddress(exampleAddress) {
+      assertTrue(false, "setRewardAddress should fail");
+    } catch {} // TODO should match on an error to make sure that this is failing in the way that's expected
+  }
+
+  function test_setEnabled() public {
+    ExampleRewardManager example = new ExampleRewardManager();
+    address exampleAddress = address(example);
+    IRewardManager rewardManager = IRewardManager(REWARD_MANAGER_ADDRESS);
+
+    assertEq(rewardManager.readAllowList(exampleAddress), 0);
+    rewardManager.setEnabled(exampleAddress);
+    assertEq(rewardManager.readAllowList(exampleAddress), 1);
+  }
+
+  function test_setRewardAddress() public {
+    ExampleRewardManager example = new ExampleRewardManager();
+    address exampleAddress = address(example);
+    IRewardManager rewardManager = IRewardManager(REWARD_MANAGER_ADDRESS);
+
+    rewardManager.setEnabled(exampleAddress);
+    example.setRewardAddress(exampleAddress);
+
+    assertEq(example.currentRewardAddress(), exampleAddress);
+  }
+
+  function test_setupReceiveFees() public {
+    ExampleRewardManager example = new ExampleRewardManager();
+    address exampleAddress = address(example);
+
+    IRewardManager rewardManager = IRewardManager(REWARD_MANAGER_ADDRESS);
+
+    rewardManager.setEnabled(exampleAddress);
+    example.setRewardAddress(exampleAddress);
+
+    exampleReceiveFees = example;
+    exampleBalance = exampleAddress.balance;
+  }
+
+  function test_receiveFees() public {
+    // noop
+  }
+
+  function test_checkReceiveFees() public {
+    assertGt(address(exampleReceiveFees).balance, exampleBalance);
```

Comment:
what is doing?

Reply:
https://github.com/ava-labs/subnet-evm/pull/601/files#r1178078510